lib/config/config.js

```diff
@@ -103,6 +104,30 @@ class Config {
       gatewayWatcher.isReady = true;
     });
   }
+
+  // this function suppose to preserve comments.
+  // due to fragile nature of YAWN it may not work properly.
+  updateGently (modifier) {
+    return new Promise((resolve, reject) => {
+      fs.readFile(this.gatewayConfigPath, 'utf8', (err, data) => {
+        if (err) {
+          return reject(err);
+        }
+        let json = yamlOrJson.load(data);
+        let result = modifier(json);
+        let text = yamlOrJson.dump(result);
+        // let yawn = new YAWN(data);
+        // yawn.json = modifier(yawn.json);
+        // let text = yaml.yaml;
+        fs.writeFile(this.gatewayConfigPath, text, (err) => {
+          if (err) {
+            return reject(err);
+          }
+          resolve();
+        });
+      });
```

Comment:
I would refactor this piece of code to take advantage of the `util.promisify` and chain the promises instead of going into a callback hell.

Something like this:

```javascript
const fs = require("fs");
const readFile = util.promisify(fs.readFile);
const writeFile = util.promisify(fs.writeFile);

Pseudo code (it might be a bit different)
function updateGently(modifier) {
  return readFile(this.gatewayConfigPath, 'utf8').then((data) => {

    let json = yamlOrJson.load(data);
    let result = modifier(json);
    let text = yamlOrJson.dump(result);
    return writeFile(this.gatewayConfigPath, text);
  });
}
```

This feels more readable to me but more importantly - easier to reason about.

Reply:
Can we consider switching to babel? async / await syntax really helps in working with promises.

Reply:
Personally speaking I would refrain to introduce babel and a transpilation step for node programs. `Promises` are good enough and can handle all the use cases without sacrificing the readability, it's just matter of reorganizing the flows a bit. We can discuss that separately.